### PR TITLE
GH-99108: Make vectorized versions of Blake2 available on x86, too

### PR DESCRIPTION
**Pull request by msprotz**
Per https://github.com/python/cpython/issues/99108#issuecomment-2401861806: trying a quick fix to see if this builds.

@gpshead can you buildbot this for Debian x86? Thanks.

If it works, I'll patch & fix upstream, then pull it in accordingly into CPython. Thank you!

* Issue: gh-99108

### CODE REVIEW

**Comment by msprotz** [comment]
(The SBOM build will fail for the time being.)

**Comment by encukou** [comment]
!buildbot x86

**Comment by msprotz** [comment]
Looks like this compiles but gives an unrelated error in tests. @gpshead @encukou do you agree? If so, I'll fix it upstream

**Comment by encukou** [comment]
Yup, I agree!
I'll need to fix that ctypes error now that it's uncovered :)

**Comment by vstinner** [comment]
> buildbot/x86 Debian Installed with X PR — Build done.
> buildbot/x86 Debian Non-Debug with X PR — Build done. 

Tests failed on these two workers, but it's a known issue: https://github.com/python/cpython/issues/121938

This PR fix the build on these two workers: https://github.com/python/cpython/issues/125535

**Review by vstinner — APPROVED** [review]
LGTM

**Comment by vstinner** [comment]
> (The SBOM build will fail for the time being.)

I ran `make regen-sbom`.

**Comment by msprotz** [comment]
The fix just got merged a couple hours ago upstream, I need to run a build to refresh the correct folder, and then I'll push the refresh here (along with make regen-sbom). Thanks!

**Comment by vstinner** [comment]
@gpshead @encukou: Why did you block merging this PR? Do you prefer to wait until upstream is fixed, and then update Hacl in Python?

**Comment by encukou** [comment]
Yes.

**Comment by msprotz** [comment]
Ok this is pushed, let's see what CI says. @vstinner FYI, this would be an SBOM violation to push a local fix to an external vendored library, so this is why we have to propagate the fix upstream first... thanks!

**Comment by vstinner** [comment]
> FYI, this would be an SBOM violation to push a local fix to an external vendored library

I didn't know that. We made many changes in mimalloc vendored copy.

**Comment by msprotz** [comment]
Maybe mimalloc gets a different treatment? The SBOM CI target fails if the vendored copy of HACL* is not exactly identical to upstream...

**Comment by gpshead** [comment]
For this one we set it up to be upstream-first and just track specific upstream versions.  it works since we've got active maintainers. :)

**Comment by zware** [comment]
!buildbot x86 Deb

**Comment by zware** [comment]
This fixes the build issue on x86 Debian.  Not sure what's up with the [CI job checking the SBOM](https://github.com/python/cpython/actions/runs/11373197759/job/31644738767?pr=125244), though.  I can't reproduce that failure locally (cc @sethmlarson for sbom insight).

File: Misc/sbom.spdx.json
Comment (zware) [inline Misc/sbom.spdx.json]: Ah ha, that's why :)

```suggestion
          "checksumValue": "935ae51d0ff0bf1403f0ecc1ff02b8f685d09053618558c07fbe4bd2abbc5dd1"
```

**Comment by vstinner** [comment]
@msprotz: Can you re-run `make regen-sbom`?

**Comment by msprotz** [comment]
I thought I had run it. Let me run it again.

**Comment by msprotz** [comment]
I committed the suggested change, but `make regen-sbom` generates no diff on my machine still. Let's see if this fixes CI

**Comment by zware** [comment]
I could have left a clearer note, sorry.  It turned out that the generated files check was failing on `regen-sbom` because `Tools/build/generate_sbom.py` behaves differently if `CI` is defined in the environment.  In that case, it actually does the download and regenerates the checksum.

**Comment by msprotz** [comment]
Ah ok! Good to know for next time, thank you.

**Pull request merged**
